inference-engine/src/plaidml_plugin/ops/assign.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+
+#include "ngraph/opsets/opset3.hpp"
+
+#include "plaidml/op/op.h"
+
+#include "plaidml/edsl/edsl.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace ngraph;
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("assign", [](const Context& ctx) {
+  auto* layer = ngraph::as_type<ngraph::opset3::Assign>(ctx.layer);
+  IE_ASSERT(ctx.operands.size() == 2);
+  auto I = ctx.operands.at(0);
+  layer->validate_and_infer_types();
+  auto variable = layer->get_variable();
```

Comment:
From assign definition, "Assign operation sets an input value to the variable_id variable and optionally returns it as an output. This value will be returned by ReadValue operation on next infer if variable was not reset. ", it seems we shall update variable and return it, not see update here